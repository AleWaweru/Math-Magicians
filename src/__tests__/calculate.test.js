import calculate from '../logic/calculate';

describe('calculate', () => {
  it('should update next number based on button name and calculator object with operation', () => {
    const obj = {
      total: 3,
      next: '4',
      operation: '+',
    };
    const buttonName = '2';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });

  it('should update next number if the button name is a decimal point', () => {
    const obj = {
      total: 2,
      next: '4',
      operation: '*',
    };
    const buttonName = '.';
    expect(calculate(obj, buttonName)).toMatchSnapshot();
  });
});
