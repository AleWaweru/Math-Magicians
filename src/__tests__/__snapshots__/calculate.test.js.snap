// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate should update next number based on button name and calculator object with operation 1`] = `
Object {
  "next": "42",
  "operation": "+",
  "total": 3,
}
`;

exports[`calculate should update next number if the button name is a decimal point 1`] = `
Object {
  "next": "4.",
  "operation": "*",
  "total": 2,
}
`;
